src/Core/AdminConsole/OrganizationFeatures/OrganizationUsers/DeleteManagedOrganizationUserAccountCommand.cs

```diff
@@ -13,87 +14,102 @@ public class DeleteManagedOrganizationUserAccountCommand : IDeleteManagedOrganiz
         var managementStatus = await _getOrganizationUsersManagementStatusQuery.GetUsersOrganizationManagementStatusAsync(organizationId, orgUserIds);
 
-        foreach (var orgUser in orgUsers)
+        foreach (var orgUserId in orgUserIds)
         {
-            if (!orgUser.UserId.HasValue)
-            {
-                results.Add((orgUser, "Invalid organization user."));
-                continue;
-            }
-
-            if (!managementStatus.TryGetValue(orgUser.Id, out var isManaged) || !isManaged)
-            {
-                results.Add((orgUser, "User is not managed by the organization."));
-                continue;
-            }
-
             try
             {
-                var userToDelete = await _userService.GetUserByIdAsync(orgUser.UserId.Value);
-                if (userToDelete == null)
+                var orgUser = orgUsers.FirstOrDefault(ou => ou.Id == orgUserId);
+                if (orgUser == null || orgUser.OrganizationId != organizationId)
                 {
-                    results.Add((orgUser, "User not found."));
-                    continue;
+                    throw new NotFoundException("Organization user not found.");
                 }
 
-                await _userService.DeleteAsync(userToDelete);
-                await _eventService.LogOrganizationUserEventAsync(orgUser, EventType.OrganizationUser_Deleted);
-                results.Add((orgUser, ""));
+                await RepositoryDeleteUserAsync(organizationId, orgUser, deletingUserId, managementStatus);
+                results.Add((orgUserId, ""));
             }
             catch (Exception e)
             {
-                results.Add((orgUser, e.Message));
+                results.Add((orgUserId, e.Message));
             }
         }
 
         return results;
     }
+
+    private async Task RepositoryDeleteUserAsync(Guid organizationId, OrganizationUser orgUser, Guid? deletingUserId, IDictionary<Guid, bool> managementStatus = null)
+    {
+        if (deletingUserId.HasValue && orgUser.UserId.Value == deletingUserId.Value)
+        {
+            throw new BadRequestException("You cannot delete yourself.");
+        }
+
+        if (!orgUser.UserId.HasValue || orgUser.Status == OrganizationUserStatusType.Invited)
+        {
+            throw new BadRequestException("You cannot delete a user with Invited status.");
+        }
+
+        if (orgUser.Type == OrganizationUserType.Owner)
+        {
+            if (deletingUserId.HasValue && !await _currentContext.OrganizationOwner(organizationId))
+            {
+                throw new BadRequestException("Only owners can delete other owners.");
+            }
+
+            if (!await _organizationService.HasConfirmedOwnersExceptAsync(organizationId, new[] { orgUser.Id }, includeProvider: true))
+            {
+                throw new BadRequestException("Organization must have at least one confirmed owner.");
+            }
```

Comment:
`HasConfirmedOwnersExceptAsync` contains a database call, this is going to hit the db once per user when it's called in a loop. Also, it doesn't work when it's called separately like this - if you're deleting all 3 owners of an organization, it will always think there's another 2 owners. It should be called once, with all the users you want to delete.